consensus/polybft/fsm.go

```diff
@@ -240,30 +267,36 @@ func (f *fsm) BuildProposal(currentRound uint64) ([]byte, error) {
 	return stateBlock.Block.MarshalRLP(), nil
 }
 
-// applyBridgeCommitmentTx builds state transaction which contains data for bridge commitment registration
-func (f *fsm) applyBridgeCommitmentTx() error {
-	if f.proposerCommitmentToRegister != nil {
-		bridgeCommitmentTx, err := f.createBridgeCommitmentTx()
-		if err != nil {
-			return fmt.Errorf("creation of bridge commitment transaction failed: %w", err)
-		}
+// applyBridgeBatchTx builds state transaction which contains data for bridge batch registration
+func (f *fsm) applyBridgeBatchTx() error {
+	for chainID, proposerBridgeBatchToRegister := range f.proposerBridgeBatchToRegister {
+		if proposerBridgeBatchToRegister != nil {
+			bridgeBatchTx, err := f.createBridgeBatchTx(chainID)
+			if err != nil {
+				return fmt.Errorf("creation of bridge batch transaction failed: %w", err)
+			}
 
-		if err := f.blockBuilder.WriteTx(bridgeCommitmentTx); err != nil {
-			return fmt.Errorf("failed to apply bridge commitment state transaction. Error: %w", err)
+			if err := f.blockBuilder.WriteTx(bridgeBatchTx); err != nil {
+				return fmt.Errorf("failed to apply bridge batch state transaction. Error: %w", err)
+			}
 		}
 	}
 
 	return nil
 }
 
-// createBridgeCommitmentTx builds bridge commitment registration transaction
-func (f *fsm) createBridgeCommitmentTx() (*types.Transaction, error) {
-	inputData, err := f.proposerCommitmentToRegister.EncodeAbi()
-	if err != nil {
-		return nil, fmt.Errorf("failed to encode input data for bridge commitment registration: %w", err)
+// createBridgeBatchTx builds bridge batch registration transaction
+func (f *fsm) createBridgeBatchTx(chainID uint64) (*types.Transaction, error) {
```

Comment:
Pass it the `commitBatchFunction` since you are already iterating through the map in the `applyBridgeTx` function, and remove this err for map without chain id and this value get from the map.